website/lib/GKDB.class.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+class GKDB {
+    private static $connectCount = 0;
+    private static $_instance = null;
+
+    private $link = null;
+
+    private function __construct() {
+    }
+
+    public static function getInstance() {
+        if (is_null(self::$_instance)) {
+            self::$_instance = new GKDB();
+        }
+
+        return self::$_instance;
+    }
+
+    public static function getLink() {
+        return self::getInstance()->connect();
+    }
+
+    public static function getConnectCount() {
+        return self::$connectCount;
+    }
+
+    public function connect() {
+        if ($this->link != null) {
+            try {
+                mysqli_get_server_info($this->link);
+
+                return $this->link;
+            } catch (Exception $exc) {
+            }
+        }
+        $dbHost = constant('CONFIG_HOST');
```

Comment:
ho no, you loose the principle of a constant ;)

Why not just use `CONFIG_HOST` where needed instead of `$dbHost`

```
this->link = mysqli_connect(CONFIG_HOST, CONFIG_USERNAME, CONFIG_PASS);
```

Small check script:
```
$ cat /tmp/test.php 
<?php
const TEST="titi";
echo TEST;

$ php /tmp/test.php
titi
```

Reply:
(thanks for overall review) 
for this one, good catch ! honestly I don't remember "WHY" :)